test/rekt/features/pingsource/features.go

```diff
@@ -191,3 +198,71 @@ func SendsEventsWithEventTypes() *feature.Feature {
 
 	return f
 }
+
+func SendsEventsWithBrokerAsSinkTLS() *feature.Feature {
+	src := feature.MakeRandomK8sName("pingsource")
+	brokerName := feature.MakeRandomK8sName("broker")
+	sinkName := feature.MakeRandomK8sName("sink")
+	triggerName := feature.MakeRandomK8sName("trigger")
+	f := feature.NewFeature()
+
+	f.Prerequisite("transport encryption is strict", featureflags.TransportEncryptionStrict())
+	f.Prerequisite("should not run when Istio is enabled", featureflags.IstioDisabled())
+
+	f.Setup("install broker", broker.Install(brokerName, broker.WithEnvConfig()...))
+	f.Setup("broker is ready", broker.IsReady(brokerName))
+	f.Setup("broker is addressable", broker.IsAddressable(brokerName))
+	f.Setup("Broker has HTTPS address", broker.ValidateAddress(brokerName, addressable.AssertHTTPSAddress))
+
+	f.Setup("install sink", eventshub.Install(sinkName, eventshub.StartReceiverTLS))
+
+	f.Setup("install trigger", func(ctx context.Context, t feature.T) {
+		d := service.AsDestinationRef(sinkName)
+		d.CACerts = eventshub.GetCaCerts(ctx)
+		trigger.Install(triggerName, brokerName, trigger.WithSubscriberFromDestination(d))(ctx, t)
+	})
+	f.Setup("Wait for Trigger to become ready", trigger.IsReady(triggerName))
+
+	f.Requirement("install PingSource", func(ctx context.Context, t feature.T) {
+		d := broker.AsDestinationRef(brokerName)
+
+		brokerAddr, err := broker.Address(ctx, brokerName)
+		if err != nil {
+			t.Fatal("failed to get the address of the broker service", brokerName, err)
+		}
+
+		d.CACerts = brokerAddr.CACerts
+
+		cfg := []manifest.CfgFn{
+			pingsource.WithSink(&duckv1.Destination{URI: brokerAddr.URL, CACerts: d.CACerts}),
+			pingsource.WithData("text/plain", "hello, world!"),
+		}
+
+		pingsource.Install(src, cfg...)(ctx, t)
+	})
+
+	f.Requirement("PingSource goes ready", pingsource.IsReady(src))
+
+	examplePodName := feature.MakeRandomK8sName("example")
+
+	// create a pod so that PingSource delivers an event to its sink
+	// event body is similar to this:
+	// {"kind":"Pod","namespace":"test-wmbcixlv","name":"example-axvlzbvc","apiVersion":"v1"}
+	f.Requirement("install example pod",
+		pod.Install(examplePodName, exampleImage,
+			pod.WithLabels(map[string]string{"e2e": "testing"})),
+	)
+
+	f.Stable("PingSource as event source").
```

Comment:
Is this feature stable? I believe we are testing with TLS, so should we be using `f.Alpha` for now?

Reply:
cc @pierDipi since I am a little unfamiliar with this area of the codebase

Reply:
This is a good point, however, we've not used `Alpha` for other tests and in our current CI pipeline it doesn't make a difference, so we can keep `Stable`, an alternative to avoid confusion is to simply use `Assert` directly

Reply:
Should we open an issue to differentiate this in our CI pipeline?

Reply:
I'd leave it as it is since for adding alpha / non blocking tests we have the experimental suite running in GH actions but I'd like to have TLS tests blocking the merge if they break (even if it's alpha) also because so far they seem pretty stable